### ISSUE TITLE
Feature request: ArgoCD dry run

### ISSUE THREAD

**Issue by sthomson-wyn** [zapier/kubechecks#160]
It would be neat if kubechecks ran a dry sync with the argocd CLI.

https://argo-cd.readthedocs.io/en/stable/user-guide/commands/argocd_app_sync/

Flags of interest:
--dry-run
--local
--server-side

If this works how I think it does, this would allow users to identify things like validating webhook violations at PR time, rather than deploy time

**Comment by djeebus** [comment]
Very interesting, I like the idea!

**Comment by MeNsaaH** [comment]
Yeah, I stumbled unto this when I was searching for how to validating webhook. I remember bumping into an issue that delayed my progress: 
1. It doesn't work for new resources, see https://github.com/argoproj/argo-cd/issues/21660
2. It sorta modifies the actual application - I need to retest this and verify though. But I noticed this wild behaviour where `argocd sync --dry-run` updated the actual app, and in most cases left that app out-of-sync. 

I'll hopefully be working on this feature soon
